lib/data/data_source/remote/authentication_remote_source_impl.dart

```diff
@@ -120,7 +123,11 @@ class AuthenticationRemoteSourceImpl implements AuthenticationRemoteSource {
             name: dto.name.isNotEmpty ? dto.name : existingUserDoc.name,
             userId: dto.userId.isNotEmpty ? dto.userId : existingUserDoc.userId,
             initialLanguage: dto.initialLanguage.isNotEmpty ? dto.initialLanguage : existingUserDoc.initialLanguage,
+            nativeLanguage: dto.nativeLanguage.isNotEmpty ? dto.nativeLanguage : existingUserDoc.nativeLanguage,
+            appLanguage: dto.appLanguage.isNotEmpty ? dto.appLanguage : existingUserDoc.appLanguage,
+            languageToLearn: dto.languageToLearn.isNotEmpty ? dto.languageToLearn : existingUserDoc.languageToLearn,
             userFolders: dto.userFolders.isNotEmpty ? dto.userFolders : existingUserDoc.userFolders,
+            email: dto.email.isNotEmpty ? dto.email : existingUserDoc.email,
```

Comment:
to not check if it's empty everytime, you can make the values nullable in the model